tensorflow/python/ops/nn_ops.py

```diff
@@ -1097,22 +1120,40 @@ def __init__(self,
     self.padding = padding
     self.name = name
     self.dilation_rate = dilation_rate
+    # We call CUDNN convolutions when dealing with convolutions with 1D/2D
+    # space + dilation or convolutions with no dilation. CUDNN is not used for
+    # 3D convolutions with dilation because that would result in "no algorithm
+    # worked" errors from CUDNN.
+    conv_dims = input_shape.ndims - 2
+    build_op = (self._build_op_atrous if fused and conv_dims <= 2 else
```

Comment:
In the "fused" case it would be better to just call _build_op_atrous directly rather than going through _WithSpaceToBatch.